src/overlays/actors/ovl_En_Gm/z_en_gm.c

```diff
@@ -20,37 +46,279 @@ const ActorInit En_Gm_InitVars = {
+    ICHAIN_U8(unk_1F, 5, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 30, ICHAIN_STOP),
+};
+
+void EnGm_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGm* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    osSyncPrintf(VT_FGCOL(GREEN) "%s[%d] : 中ゴロン[%d]" VT_RST "\n", "../z_en_gm.c", 133,
+                 this->actor.params); // "%s[%d]: Medi Goron [%d]"
+    this->objBankIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_GM);
+    if (this->objBankIndex < 0) {
+        osSyncPrintf(VT_COL(RED, WHITE));
+        osSyncPrintf("モデル バンクが無いよ！！（中ゴロン）\n"); // "There is no model bank! !! (Medi Goron)"
+        osSyncPrintf(VT_RST);
+        __assert("0", "../z_en_gm.c", 145);
+    }
+    this->actionFunc = func_80A3D838;
+}
+
+void EnGm_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnGm* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+s32 func_80A3D7C8(void) {
+    if (LINK_AGE_IN_YEARS == YEARS_CHILD) {
+        return 0;
+    }
+
+    if (!(gBitFlags[2] & gSaveContext.equipment)) {
+        return 1;
+    }
+
+    if (gBitFlags[3] & gSaveContext.equipment) {
+        return 2;
+    }
+    return 3;
+}
+
+void func_80A3D838(EnGm* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objBankIndex)) {
+        this->actor.flags &= ~0x10;
+        SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600FEF0, NULL, this->limbDrawTable, this->transitionDrawTable,
+                         18);
+        gSegments[6] = VIRTUAL_TO_PHYSICAL(globalCtx->objectCtx.status[this->objBankIndex].segment);
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060002B8, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_060002B8.genericHeader), 0, 0.0f);
+        this->actor.draw = EnGm_DrawProc;
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder_Set3(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 35.0f);
+        Actor_SetScale(&this->actor, 0.05f);
+        this->actor.colChkInfo.mass = 0xFF;
+        this->eyeIndex = 0;
+        this->eyeTimer = 20;
+        this->actor.textId = 0x3049; // "Tight fit..."
+        this->actionFunc = func_80A3DFBC;
+        this->subActionFunc = func_80A3DB04;
+        this->actor.speedXZ = 0.0f;
+        this->actor.gravity = -1.0f;
+        this->actor.velocity.y = 0.0f;
+    }
+}
+
+void EnGm_UpdateEye(EnGm* this) {
+    if (this->eyeTimer != 0) {
+        this->eyeTimer--;
+    } else if ((u8)(this->eyeIndex += 1) >= 3) {
+        this->eyeIndex = 0;
+        this->eyeTimer = Math_Rand_ZeroFloat(60.0f) + 20.0f;
+    }
+}
+
+void EnGm_SetTextID(EnGm* this) {
+    switch (func_80A3D7C8()) {
+        case 0:
+            if (gSaveContext.infTable[11] & 1) {
+                this->actor.textId = 0x304B; // "If you can wait five or six years, it should be ready. OK?"
```

Comment:
Fig [mentioned](https://github.com/zeldaret/oot/pull/380#discussion_r491707947) that we shouldn't include text excerpts anymore